department-of-veterans-affairs/va.gov-cms
# VACMS-3998: user import improvements

## Description

See #3998 

## Testing done

Manual testing on lando with default env and faked as BRD


## QA steps

As an admin user:
1. Go to `/migrate_source_ui`
1. Import a file with emails that are @va.gov and another domain
1. Then validate Acceptance Criteria from issue
- [x] VA emails are imported
- [x] Non-VA emails are not imported and you see this message:
  - [x] "The user with email address foo@example.com was not created, since it is not a va.gov email address."
- [x] Password is set to `drupal8` and the user can log in with password
- [x] These messages are shown:
  - [x] "All accounts are created as active."
  - [x] "Password was set to drupal8 for all imported users"

Then, fake the environment you're using to test as BRD by editing its settings file:
- Change `$settings['va_gov_frontend_build_type']` to `brd`.
- Add this line: `settings['jenkins_build_env'] = 'test';`

1. Then validate Acceptance Criteria from issue
- [x] VA emails are imported
- [x] Non-VA emails are not imported and you see this message:
  - [x] "The user with email address foo@example.com was not created, since it is not a va.gov email address."
- [x] New user accounts should be configured to use SSO for login
- [x] New user accounts are blocked
- [x] These messages are shown:
  - [x] "All accounts are created as blocked."
  - [x] "Password was set to drupal8 for all imported users"

### Definition of Done


### Select Team for PR review


### Is this PR blocked by another PR?


### Does this PR need review from a Product Owner


### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
